packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -864,6 +878,177 @@ library ConnextLogic {
+      revert ConnextLogic__reconcile_invalidAction();
+    }
+
+    // load the transferId
+    bytes32 transferId = action.transferId();
+
+    // ensure the transaction has not been handled
+    if (_reconciledTransfers[transferId]) {
+      revert ConnextLogic__reconcile_alreadyReconciled();
+    }
+
+    // get the token contract for the given tokenId on this chain
+    // (if the token is of remote origin and there is
+    // no existing representation token contract, the TokenRegistry will
+    // deploy a new one)
+    address token = _args.tokenRegistry.ensureLocalToken(tokenId.domain(), tokenId.id());
+
+    // load amount once
+    uint256 amount = action.amnt();
+
+    // NOTE: tokenId + amount must be in plaintext in message so funds can
+    // *only* be minted by `handle`. They are still used in the generation of
+    // the transferId so routers must provide them correctly to be reimbursed
+
+    // TODO: do we need to keep this
+    bytes32 details = action.detailsHash();
+
+    // if the token is of remote origin, mint the tokens. will either
+    // - be credited to router (fast liquidity)
+    // - be reserved for execution (slow liquidity)
+    if (!_args.tokenRegistry.isLocalOrigin(token)) {
+      IBridgeToken(token).mint(address(this), amount);
+      // Tell the token what its detailsHash is
+      IBridgeToken(token).setDetailsHash(details);
+    }
+    // NOTE: if the token is of local origin, it means it was escrowed
+    // in this contract at xcall
+
+    // mark the transfer as reconciled
+    _reconciledTransfers[transferId] = true;
+
+    return (amount, token, transferId);
+  }
+
+  /**
+   * @notice Repays to Aave Portal if the transfer was executed with fast path using Portal liquidity
+   * @return The amount left after the repayment
+   * @return The address of the adopted asset
+   */
+  function _reconcileProcessPortal(
+    ReconcileArgs memory _args,
+    uint256 _amount,
+    address _token,
+    bytes32 _transferId,
+    mapping(bytes32 => uint256) storage _aavePortalsTransfers,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => address) storage _canonicalToAdopted
+  ) private returns (uint256, address) {
+    ReconcilePortalVars memory vars;
+
+    vars.portalTransferAmount = _aavePortalsTransfers[_transferId];
+
+    if (vars.portalTransferAmount == 0) return (_amount, _token);
+
+    // Swap all the transfer amount from local asset to adopted if needed
+    (vars.adoptedAmountOut, vars.adopted) = AssetLogic.swapFromLocalAssetIfNeeded(
+      _canonicalToAdopted,
+      _adoptedToLocalPools,
+      _args.tokenRegistry,
+      _token,
+      _amount
+    );
+
+    // Calculates the amount to be repaid to the portal
+    (vars.totalRepayAmount, vars.backUnbackedAmount, vars.portalFee) = _calculatePortalRepayment(
+      vars.adoptedAmountOut,
+      vars.portalTransferAmount,
+      _args.portalFeeNumerator,
+      _args.portalFeeDenominator
```

Comment:
where do these get assigned?

Reply:
These get assigned in the handle method before calling ConnextLogic here https://github.com/connext/nxtp/blob/71e7d7af0ac2111779c92dcbc33e64710a66528c/packages/deployments/contracts/contracts/nomad-xapps/contracts/connext/ConnextHandler.sol#L480-L481